presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -126,6 +126,7 @@
     private final AtomicBoolean wasNull = new AtomicBoolean();
     protected final AtomicBoolean closed = new AtomicBoolean();
     private final Optional<Statement> statement;
+    protected int currentRow; // Index into 'rows' of our current row (1-based)
```

Comment:
Could you move to after `AtomicReference<List<Object>> row`? These variables related each other.
```suggestion
    private final AtomicInteger currentRow = new AtomicInteger(); // Index into 'rows' of our current row (1-based)
```

Reply:
i know it's preexisting, but i don't know why we use atomics for ResultSet state.
Does RS need to be thread safe? @electrum?

even then, using `synchronized` + `@GuardedBy` could yield to simpler code.

(no change requested here within this PR)